gapic-common/lib/gapic/rest/grpc_transcoder.rb

```diff
@@ -182,9 +181,14 @@ def construct_body_query_params body_template, request_hash_without_uri, request
           #
           # The `request_hash_without_uri` at this point was mutated to delete these fields.
           #
-          # Note that the body template can only point to a top-level field
-          request_hash_without_uri.delete camel_name_for body_template
-          body = request.send(body_template.to_sym).to_json(emit_defaults: true)
+          # Note 1: body template can only point to a top-level field.
+          # Note 2: The field that body template points to can be null, in which case
+          # an empty string should be sent. E.g. `Compute.Projects.SetUsageExportBucket`.
+          if request.respond_to?(body_template.to_sym) && request.send(body_template.to_sym)
+            request_hash_without_uri.delete camel_name_for body_template
+            body = request.send(body_template.to_sym).to_json(emit_defaults: true)
```

Comment:
Optional style nit: The repeating of `request.send(body_template.to_sym)` is a bit awkward and makes this a bit hard to read. Consider setting it to a temp variable like `raw_body` (after verifying respond_to of course).